### ISSUE TITLE
firefox 92.0: crash at displaying GTK file dialogue

### ISSUE THREAD

**Issue by schmittlauch** [NixOS/nixpkgs#140909]
### Describe the bug
When attempting to display a GTK file open dialogue, Firefox crashes and coredumps.
Possible szenarios when this dialog is being displayed are a file picker form field in a website, or the "File -> Open File" menu. 

### Steps To Reproduce
1. Open Firefox
2. Press the "Alt" key to display the menu bar
3. Select "File -> Open File"
4. See Firefox crashing and dumping core.

### Additional context

Backtrace obtained from the coredump:

```bt
#0  0x00007f23bd8cc507 in g_log_structured_array () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libglib-2.0.so.0
#1  0x00007f23bd8cc915 in g_log_default_handler () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libglib-2.0.so.0
#2  0x00007f23bd8ccb5e in g_logv () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libglib-2.0.so.0
#3  0x00007f23bd8cce0f in g_log () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libglib-2.0.so.0
#4  0x00007f23bd73220a in g_settings_set_property () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgio-2.0.so.0
#5  0x00007f23bd82e936 in object_set_property () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#6  0x00007f23bd82ecfc in g_object_new_internal () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#7  0x00007f23bd830618 in g_object_new_valist () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#8  0x00007f23bd830b49 in g_object_new () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#9  0x00007f23bdc3be4c in _gtk_file_chooser_get_settings_for_widget () from /nix/store/bsc3xi2hrhrkd40vyfb078kx60jsbldx-gtk+3-3.24.27/lib/libgtk-3.so.0
#10 0x00007f23bdc3ce40 in gtk_file_chooser_widget_get_default_size () from /nix/store/bsc3xi2hrhrkd40vyfb078kx60jsbldx-gtk+3-3.24.27/lib/libgtk-3.so.0
#11 0x00007f23bdc373b6 in file_chooser_widget_default_size_changed () from /nix/store/bsc3xi2hrhrkd40vyfb078kx60jsbldx-gtk+3-3.24.27/lib/libgtk-3.so.0
#12 0x00007f23bd8295cf in g_closure_invoke () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#13 0x00007f23bd83b28b in signal_emit_unlocked_R.isra.0 () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#14 0x00007f23bd84196f in g_signal_emit_valist () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#15 0x00007f23bd8420ef in g_signal_emit_by_name () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#16 0x00007f23bd82e936 in object_set_property () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#17 0x00007f23bd830ccc in g_object_setv () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#18 0x00007f23bd8319ab in g_object_set_property () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#19 0x00007f23bd82e936 in object_set_property () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#20 0x00007f23bd830fa9 in g_object_set_valist () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#21 0x00007f23bd83187c in g_object_set () from /nix/store/a4l1kcvgicllc0k6lp5salmk2nx2gwxa-glib-2.68.4/lib/libgobject-2.0.so.0
#22 0x00007f23b8042c91 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#23 0x00007f23b4a29232 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#24 0x00007f23b639e1c8 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#25 0x00007f23b639efae in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#26 0x00007f23b8fbca7d in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#27 0x00007f23b8fb5f4a in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#28 0x00007f23b8fadd62 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#29 0x00007f23b8fbcbbb in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#30 0x00007f23b8fbd2c7 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#31 0x00007f23b907ec94 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#32 0x00007f23b71c8a9e in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#33 0x00007f23b76318a3 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#34 0x00007f23b761c96a in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#35 0x00007f23b761cfd1 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#36 0x00007f23b7615d18 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#37 0x00007f23b761555d in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#38 0x00007f23b76173fe in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#39 0x00007f23b7619444 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#40 0x00007f23b6a75c10 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#41 0x00007f23b7620e6b in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#42 0x00007f23b68bfedd in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#43 0x00007f23b7e39c78 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#44 0x00007f23b7616aff in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#45 0x00007f23b7619444 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#46 0x00007f23b81c9e42 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#47 0x00007f23b68bfebe in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#48 0x00007f23b83ebc93 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#49 0x00007f23b5caeabe in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#50 0x00007f23b5c9b76a in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#51 0x00007f23b5c9a0c7 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#52 0x00007f23b5c9a308 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#53 0x00007f23b5cb0582 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#54 0x00007f23b5ca5592 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#55 0x00007f23b5ca98d6 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#56 0x00007f23b60a3ca8 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#57 0x00007f23b60668c8 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#58 0x00007f23b7fb96d9 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#59 0x00007f23b8e67d12 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#60 0x00007f23b8f128dc in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#61 0x00007f23b8f12f45 in ?? () from /nix/store/sqj9f30kk2yb2zidnsdzn06q5l1n25mw-firefox-92.0/lib/firefox/libxul.so
#62 0x00005556f1d98234 in ?? ()
#63 0x00007f23bee89ded in __libc_start_main () from /nix/store/jsp3h3wpzc842j0rz61m5ly71ak6qgdn-glibc-2.32-54/lib/libc.so.6
#64 0x00005556f1d97a2a in _start ()
```

If necessary, I can try to obtain a backtrace again with debug symbols.

### Notify maintainers
@edolstra @lovesegfault @mweinelt

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.70, NixOS, 21.05.3532.aff647e2704 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.05.3532.aff647e2704, nixos-unstable-21.11pre321003.14aef06d9b3"`
 - channels(user): `"NUR, home-manager-21.05, nixos-21.05.3532.aff647e2704"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: firefox
# a list of nixos modules affected by the problem
module:
```

**Comment by jtojnar** [comment]
Do you have any errors in the console/system journal when that happens?

**Comment by schmittlauch** [comment]
There were no errors in the console output of Firefox. But this issue appears to be fixed in NixOS 21.05.4132.24528474d2.
